src/dll.rs

```diff
@@ -0,0 +1,195 @@
+/// A simple doubly-linked list for use in the `Lru`
```

Comment:
this is the main thing that powers an `Lru` now, protected by a mutex and heavily sharded, so that the threads are using like 256 of these for now to avoid contention.